Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -1591,29 +1598,68 @@ - (void)handleInitSuccess {
+            
+            
+            BranchDeepLinkModel* deepLinkModel = self.deepLinkControllers[key];
+            UIViewController <BranchDeepLinkingController> *branchSharingController = deepLinkModel.viewController;
+            
+            
             if ([branchSharingController respondsToSelector:@selector(configureControlWithData:)]) {
                 [branchSharingController configureControlWithData:latestReferringParams];
             }
             else {
                 [self.preferenceHelper log:FILE_NAME line:LINE_NUM message:
-                    @"[Branch Warning] View controller does not implement configureControlWithData:"];
+                 @"[Branch Warning] View controller does not implement configureControlWithData:"];
             }
             branchSharingController.deepLinkingCompletionDelegate = self;
             self.deepLinkPresentingController = [[[UIApplicationClass sharedApplication].delegate window] rootViewController];
             
-            if ([self.deepLinkPresentingController presentedViewController]) {
-                [self.deepLinkPresentingController dismissViewControllerAnimated:NO completion:^{
-                    [self.deepLinkPresentingController presentViewController:branchSharingController animated:YES completion:NULL];
-                }];
-            }
-            else {
-                [self.deepLinkPresentingController presentViewController:branchSharingController animated:YES completion:NULL];
+            
+            switch (deepLinkModel.option) {
+                case BNCViewControllerOptionPresent:
+                    [self presentSharingViewController:branchSharingController];
+                    break;
+                    
+                case BNCViewControllerOptionPush:
+                    
+                    if ([self.deepLinkPresentingController isKindOfClass:[UINavigationController class]]) {
+                        [(UINavigationController*)self.deepLinkPresentingController pushViewController:branchSharingController animated:true];
+                    }else{
+                        [self presentSharingViewController:branchSharingController];
+                    }
+                    
+                    NSAssert([self.deepLinkPresentingController isKindOfClass:[UINavigationController class]]==NO, @"Root View Controller is not of type UINavigationController, hence cannot push");
+                    
+                    break;
+                    
+                default:
+                    if ([self.deepLinkPresentingController isKindOfClass:[UINavigationController class]]) {
+                        if ([self.deepLinkPresentingController respondsToSelector:@selector(showViewController:sender:)]) {
+                            [self.deepLinkPresentingController showViewController:branchSharingController sender:self];
+                        }else{
```

Comment:
Can you use spaces as such `} else {` to keep with rest of styling. I see it a couple times throughout the project